cogs/moderation.py

```diff
@@ -20,91 +129,216 @@ def __init__(self, bot: Bot) -> None:
+                **Actor**: {ctx.author.mention} (`{ctx.author.id}`)
+                """
+            ),
+            color=discord.Color.orange(),
+            timestamp=datetime.utcnow(),
+        )
+        server_embed.set_thumbnail(url=member.avatar_url_as(format="png", size=256))
 
-        await ctx.send(embed=embed)
-        await member.send(embed=embed)
+        dm_embed = discord.Embed(
+            title="You were Banned",
+            description=textwrap.dedent(
+                f"""
+                {reason}
+
+                *Server: {ctx.guild.name}*
+                """
+            ),
+            color=discord.Color.red(),
+            timestamp=datetime.utcnow(),
+        )
+        dm_embed.set_thumbnail(url=ctx.guild.icon_url)
+
+        await ctx.send(embed=server_embed)
+        await member.send(embed=dm_embed)
         await member.ban(reason=reason)
 
     @command()
-    @bot_has_permissions(ban_members=True)
     @has_permissions(ban_members=True)
-    async def unban(ctx: Context, *, member: str) -> None:
-        """Unban a User."""
-        # TODO: Use custom converter for `member`
-        banned_users = await ctx.guild.bans()
-        member_name, member_discriminator = member.split("#")
+    async def multiban(self, ctx: Context, members: Greedy[UserID], *, reason: str = None) -> None:
+        """Bans multiple members from the server."""
+        if reason is None:
+            reason = f"Action done by {ctx.author} (ID: {ctx.author.id})"
+
+        total_members = len(members)
+        if total_members == 0:
+            return await ctx.send("No members to ban.")
 
-        # TODO: There might be a better way to handle this
-        for ban_entry in banned_users:
-            user = ban_entry.user
+        confirm = await ctx.prompt(f"This will ban **{Plural(total_members):member}**. Are you sure?", reacquire=False)
+        if not confirm:
+            return await ctx.send("Aborting.")
 
-            if (user.name, user.discriminator) == (member_name, member_discriminator):
-                await ctx.guild.unban(user)
-                await ctx.send(f"Unbanned **{user.name}#{user.discriminator}**")
-                return
+        failed = 0
+        for member in members:
+            try:
+                await ctx.guild.ban(member, reason=reason)
+            except discord.HTTPException:
+                failed += 1
+
+        await ctx.send(f"Banned {total_members - failed}/{total_members} members.")
+
+    @command()
+    @has_permissions(ban_members=True)
+    async def unban(self, ctx: Context, *, user: UserID) -> None:
+        """Unban a User."""
+        try:
+            await ctx.guild.unban(user)
+
+            embed = discord.Embed(
+                title="User Unbanned",
+                description=textwrap.dedent(
+                    f"""
+                    **User**: {user.mention} (`{user.id}`)
+                    **Actor**: {ctx.author.mention} (`{ctx.author.id}`)
```

Comment:
```suggestion
                    **Moderator**: {ctx.author.mention} (`{ctx.author.id}`)
```

Reply:
Fixed in cae3d0229dfe10db6f1ebd9ecdde6bc98c5f61d7